### PR TITLE
add-cgroupv1-support

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/28560

### CODE REVIEW

**Comment by marians** [comment]
I think the common term is "cgroups" with an `s` in the end, so I'd recommend to change cgroupv1 to cgroupsv1 etc. everywhere.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Allow cluster-autoscaler handling MachinePools.
 - Add additional tag for cluster autoscaler to MachinePool ASGs.
+- Add option to force CGroup v1.
```
Comment (marians) [inline CHANGELOG.md:16]:
```suggestion
- Add option to force CGroups v1.
```

File: helm/cluster-aws/README.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ Properties within the `.providerSpecific` top-level object
 | `providerSpecific.additionalResourceTags.*` | **Tag value**|**Type:** `string`<br/>**Value pattern:** `^[ a-zA-Z0-9\._:/=+-@]+$`<br/>|
 | `providerSpecific.ami` | **Amazon machine image (AMI)** - If specified, this image will be used to provision EC2 instances.|**Type:** `string`<br/>|
 | `providerSpecific.awsClusterRoleIdentityName` | **Cluster role identity name** - Name of an AWSClusterRoleIdentity object. This in turn refers to the IAM role used to create all AWS cloud resources when creating the cluster. The role can be in another AWS account in order to create all resources in that account. Note: This name does not refer directly to an IAM role name/ARN.|**Type:** `string`<br/>**Value pattern:** `^[-a-zA-Z0-9_\.]{1,63}$`<br/>**Default:** `"default"`|
+| `providerSpecific.cgroupsv1` | **CGroup v1** - Force use of CGroup v1 for whole cluster.|**Type:** `boolean`<br/>**Default:** `false`|
```
Comment (marians) [inline helm/cluster-aws/README.md:24]:
```suggestion
| `providerSpecific.cgroupsv1` | **CGroups v1** - Force use of CGroups v1 for whole cluster.|**Type:** `boolean`<br/>**Default:** `false`|
```

File: helm/cluster-aws/values.schema.json [hunk 3]
```diff
@@ -950,6 +950,12 @@
                     "minLength": 1,
                     "pattern": "^[-a-zA-Z0-9_\\.]{1,63}$"
                 },
+                "cgroupsv1": {
+                    "type": "boolean",
+                    "title": "CGroup v1",
```
Comment (marians) [inline helm/cluster-aws/values.schema.json:955]:
```suggestion
                    "title": "CGroups v1",
```

File: helm/cluster-aws/values.schema.json [hunk 4]
```diff
@@ -950,6 +950,12 @@
...
+                    "description": "Force use of CGroup v1 for whole cluster.",
```
Comment (marians) [inline helm/cluster-aws/values.schema.json:956]:
```suggestion
                    "description": "Force use of CGroups v1 for whole cluster.",
```

**Comment by calvix** [comment]
> I think the common term is "cgroups" with an `s` in the end, so I'd recommend to change cgroupv1 to cgroupsv1 etc. everywhere.

I adjusted the naming But I don't think that is really true. It used to be called `cgroups` when you talk about the whole thing(both v1 and v2) but if the version is named then I never saw it written with `s`

example: https://kubernetes.io/docs/concepts/architecture/cgroups/#check-cgroup-version

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**